src/main/scala/sangria/slowlog/OpenTracing.scala

```diff
@@ -0,0 +1,53 @@
+package sangria.slowlog
+
+
+import io.opentracing.{Span, Tracer}
+import sangria.execution._
+import sangria.schema.Context
+
+import scala.collection.concurrent.TrieMap
+
+class OpenTracing(implicit private val tracer: Tracer)
```

Comment:
Also, do you think it would be a good idea to add another optional parameter for a parent `Span` for the whole thing? So that you can use it as a parent in `beforeQuery` if needed.

Reply:
Adding it as optional parameter to the middleware would require recreating the middleware every time we receive a request. Not sure would be the best thing to do.

The approach i'm using in my project is "activate" the parent span just before calling `Executor.execute`, then the query span in `beforeQuery` by having nothing specified will pick up that span as parent.